This error wasn't intended, but it confirms that the fixes for testing `CMAKE_MATCH_<...>` were successful:
~~~
CMake Error at /mnt/vcpkg-ci/installed/x64-linux/share/unit-test-cmake/test-macros.cmake:19 (_message):
   string(REGEX MATCH "--host=[^;]*-([^-;]*-[^-;]*)" output "--host=armv7a-linux-androideabi")  resulted in the wrong value for CMAKE_MATCH_1;
      expected: "linux-android"
      actual  : "linux-androideabi"
~~~

--- a/scripts/test_ports/vcpkg-ci-vcpkg-make/test-z_vcpkg_make_get_configure_triplets.cmake
+++ b/scripts/test_ports/vcpkg-ci-vcpkg-make/test-z_vcpkg_make_get_configure_triplets.cmake
@@ -42,7 +42,7 @@ if(VCPKG_HOST_IS_WINDOWS)
             CMAKE_MATCH_1 "unknown-mingw32"
         )
     else()
-        # maybe crosscompiling
+        # maybe there is --host
     endif()
 elseif(NOT VCPKG_CROSSCOMPILING)
     if(triplets)
@@ -58,7 +58,7 @@ elseif(NOT VCPKG_CROSSCOMPILING)
     endif()
 elseif(VCPKG_TARGET_IS_ANDROID)
     unit_test_check_variable_equal(
-        [[ string(REGEX MATCH "--host=[^;]*-([^-;]*-[^-;]*)" output "${triplets}") ]]
+        [[ string(REGEX MATCH "--host=[^;]*-([^-;]*-(android|[^-;]*))" output "${triplets}") ]]
         CMAKE_MATCH_1 "linux-android"
     )
     # expected bad: there is --host but not --build
@@ -68,7 +68,7 @@ elseif(VCPKG_TARGET_IS_ANDROID)
     )
 elseif(VCPKG_TARGET_IS_MINGW)
     unit_test_check_variable_equal(
-        [[ string(REGEX MATCH "--host=[^;]*-([^-;]*)" output "${triplets}") ]]
+        [[ string(REGEX MATCH "--host=[^;]*-(mingw32|[^-;]*)" output "${triplets}") ]]
         CMAKE_MATCH_1 "mingw32"
     )
     # expected bad: there is --host but not --build

